src/Scarf/Gateway/Rule.hs

```diff
@@ -560,20 +561,43 @@ matchFlatfile FlatfileRule {..} request@Request {requestPath, requestQueryString
                 ( RedirectTo
                     ( FlatfileCapture
                         { filePackage = rulePackage,
-                          fileAbsoluteUrl = absoluteUrl,
+                          fileAbsoluteUrl = Just absoluteUrl,
                           fileVariables = extracted
                         }
                     )
                     (Text.encodeUtf8 expanded)
                 )
             )
+      | Nothing <- ruleBackendTemplate,
+        let extracted = HashMap.fromList xs,
+        HashSet.isSubsetOf ruleExpectedVariables (HashMap.keysSet extracted) ->
```

Comment:
I think this can go, in case there is no backend url there is no reason to check if we have all the variables.
```suggestion
      | Nothing <- ruleBackendTemplate,
        let extracted = HashMap.fromList xs ->
```